specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -401,65 +396,68 @@ string launchArguments);</td>
-            }
-    }
+```c#
+public class App {
+
+        // App was just launched
+        protected override void OnLaunched(LaunchActivatedEventArgs args)
+        {
+            // If the application was terminated for example and has now been restarted
+            if (args != NULL)
+            {
+                if (args.PreviousExecutionState == ApplicationExecutionState.Terminated)
+                {
+                    Telemetry::WriteLine("Restart activated: " + args.Arguments);
+                }
+            }
+        }
+
+        // We want to register the app for restart if termination happens only after crash
+        protected bool restartOnUpdate(){
+            bool result = AppLifecycle::RegisterForRestart("args for new instance", RestartContext::Crash);
+            Telemetry::WriteLine("RegisterForRestart: Restart Context: " + RestartContext::Crash.ToString() 
+            + ". Result: " + result ? "success" : "fail");
+
+            // Do normal app behavior
+        }
+
+        // We want to unregister for post-crash restarts
+        protected bool removeRestartOnUpdate(){
+            bool result = AppLifecycle::UnregisterForRestart();
+            Telemetry::WriteLine("UnregisterForRestart: Result: " + result ? "success" : "fail");
+        }
+}
+```
 
 ### RequestRestartNow Sample
 
-    public class App {
-           // Let's assume this method is run when an app updates assets.
-           // The new assets have downloaded and installed, and now the app has to restart immediately
-           protected void updateInstallComplete(){
-               // code here that checks the app update has installed
-
-               WCHAR buffer\[1024\]; 
-               AppRestartFailureReason result = AppLifecycle::RequestRestartNow("args for new instance");
-
-               switch (result){
-                    case AppRestartFailureReason::RestartPending:
-                        Telemetry::WriteLine("Restart request successful.");
-                        break;
-                    case AppRestartFailureReason::NotInForeground:
-                        wcscpy_s(buffer, L"App must be in the foreground to request restart.");
-                        break;
-                    case AppRestartFailureReason::InvalidUser:
-                        wcscpy_s(buffer, L"Cannot restart for this user.");
-                        break;
-                    case AppRestartFailureReason::Other:
-                        wcscpy_s(buffer, L"Unknown failure.");
-                        break;
-               }
-           }               
-    }  
+```c#
+public class App {
+       // Let's assume this method is run when an app updates assets.
+       // The new assets have downloaded and installed, and now the app has to restart immediately
+       protected void updateInstallComplete(){
+           // code here that checks the app update has installed
+
+           WCHAR buffer\[1024\]; 
```

Comment:
NIT: `WCHAR buffer[1024]{};`

[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es23-prefer-the--initializer-syntax)